### PR TITLE
Introduce fedora eln image and ISO for anaconda

### PR DESCRIPTION
**Pull request by KKoukiou**
Image refresh for fedora-eln-boot
 * [x] image-refresh fedora-eln-boot
 * [x] image-refresh fedora-eln
 * [x] image-refresh fedora-39

Being used in https://github.com/rhinstaller/anaconda-webui/pull/70

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh fedora-eln-boot done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-boot-20231123-130843

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-boot-20231123-130808.log

**Comment by KKoukiou** [comment]
Tests are still running on rawhide. Needs https://github.com/rhinstaller/anaconda-webui/pull/63

**Comment by jkonecny12** [comment]
Hi, @sgallagher, FYI this might be interesting to you.

**Comment by cockpituous** [comment]
image-refresh fedora-eln-boot done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-boot-20231124-171612

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-boot-20231124-171616.log

**Comment by cockpituous** [comment]
image-refresh fedora-eln done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-20231124-172055

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231124-172020.log

File: images/scripts/fedora-eln-boot.bootstrap [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (martinpitt) [inline images/scripts/fedora-eln-boot.bootstrap:1]: Please also add these images to image-trigger.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! Setting the blocked label as per your comment above.

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231127-103030.log

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231127-130404.log

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231127-140000.log

**Comment by sgallagher** [comment]
@KKoukiou The `xmlto` package should be in the CRB (see https://github.com/fedora-eln/eln/issues/172) and I'm fixing that now. However, the `nodejs-devel` package is currently not available on RHEL in any public repository. It exists only in the Koji/Brew buildroot repositories. If you need that, you will want to enable the repo at `https://kojipkgs.fedoraproject.org/repos/eln-build/latest/$basearch/`

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231127-162222.log

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231128-100101.log

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231128-104646.log

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231128-105959.log

File: images/scripts/fedora.setup [hunk 2]
```diff
@@ -190,32 +194,34 @@ fi
 
 # don't use a bootstrap chroot to save time and space; everything is offline, and we always just do exactly one build
 echo "config_opts['use_bootstrap'] = False" >>/etc/mock/site-defaults.cfg
-su builder -c "/usr/bin/mock --no-bootstrap-chroot --verbose -i $(/var/lib/testvm/build-deps.sh "${ID} ${VERSION_ID}")"
+if [ "fedora-eln" != "$IMAGE" ]; then
```
Comment (martinpitt) [inline images/scripts/fedora.setup:197]: This is now an ugly special case. I'd rather you taught build-deps.sh to handle eln properly, with `fedora*eln) ;;` You almost had it correct before, but the argument is not separated with `-` but with ` `

Comment (martinpitt) [reply images/scripts/fedora.setup:197]: ... and I suppose you still actually do want to build stuff on the eln image, so you still need build-deps.sh?

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231128-114141.log

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231128-115555.log

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231128-125353.log

**Comment by cockpituous** [comment]
image-refresh fedora-eln done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-20231129-141759

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231129-141717.log

File: images/scripts/fedora.setup [hunk 3]
```diff
@@ -190,22 +201,31 @@ fi
 
 # don't use a bootstrap chroot to save time and space; everything is offline, and we always just do exactly one build
 echo "config_opts['use_bootstrap'] = False" >>/etc/mock/site-defaults.cfg
-su builder -c "/usr/bin/mock --no-bootstrap-chroot --verbose -i $(/var/lib/testvm/build-deps.sh "${ID} ${VERSION_ID}")"
+su builder -c "/usr/bin/mock --no-bootstrap-chroot --verbose -i $(/var/lib/testvm/build-deps.sh "${ID} ${VERSION_ID}" "${ID} ${VARIANT_ID}")"
```
Comment (martinpitt) [inline images/scripts/fedora.setup:204]: Why does it pass $ID twice?

File: images/scripts/lib/build-deps.sh [hunk 4]
```diff
@@ -46,7 +47,8 @@ esac
 
 # pull nodejs-devel on Fedora for compliance with the guidelines on using nodejs modules:
 # https://docs.fedoraproject.org/en-US/packaging-guidelines/Node.js/#_buildrequires
-case "$OS_VER" in
+case "$OS_VARIANT" in
```
Comment (martinpitt) [inline images/scripts/lib/build-deps.sh:50]: Ah, that's the "pass twice" bit. But please keep the VERsion here as well, we needed it in the past. I suggest to pass a single "$ID $VERSION_ID $VARIANT_ID" argument and match on that.

File: images/scripts/rhel-7-9.setup [hunk 5]
```diff
@@ -236,7 +236,7 @@ if [ "$DO_BUILD" -eq 1 ]; then
     useradd -c Builder -G mock builder
     opsys=$(cut -d '-' -f 1 <<< "$IMAGE")
     version=$(cut -d '-' -f 2 <<< "$IMAGE")
-    su builder -c "/usr/bin/mock --verbose -i $(/var/lib/testvm/build-deps.sh "$opsys $version")"
+    su builder -c "/usr/bin/mock --verbose -i $(/var/lib/testvm/build-deps.sh "$opsys $version" "$opsys")"
```
Comment (martinpitt) [inline images/scripts/rhel-7-9.setup:239]: ... then this can go away

File: images/scripts/rhel.setup [hunk 6]
```diff
@@ -328,7 +328,7 @@ EOF
     useradd -c Builder -G mock builder
     opsys=$(cut -d '-' -f 1 <<< "$IMAGE")
     version=$(cut -d '-' -f 2 <<< "$IMAGE")
-    su builder -c "/usr/bin/mock --verbose --no-bootstrap-image -i $(/var/lib/testvm/build-deps.sh "$opsys $version")"
+    su builder -c "/usr/bin/mock --verbose --no-bootstrap-image -i $(/var/lib/testvm/build-deps.sh "$opsys $version" "$opsys")"
```
Comment (martinpitt) [inline images/scripts/rhel.setup:331]: and this also

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! As this is quite intrusive, let's also rebuild fedora-39 once you are done with your changes.

**Comment by KKoukiou** [comment]
Anaconda Web UI tests on the fedora ELN here: https://cockpit-logs.us-east-1.linodeobjects.com/pull-70-20231129-160620-4fec1915-fedora-eln-boot-bots-5585/log.html#29-2

Some failures because of missing btrfs support and missing blivet-gui.

Adding skipImage now.

**Comment by cockpituous** [comment]
image-refresh fedora-eln-boot done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-boot-20231129-175014

**Comment by cockpituous** [comment]
image-refresh fedora-eln done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-20231129-180554

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231129-180505.log

**Comment by cockpituous** [comment]
image-refresh fedora-eln done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-20231129-180732

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231129-180707.log

**Comment by cockpituous** [comment]
image-refresh fedora-39 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-39-20231129-183046

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-39-20231129-183030.log

**Comment by cockpituous** [comment]
image-refresh fedora-39 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-39-20231129-183239

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-39-20231129-183232.log

**Comment by cockpituous** [comment]
image-refresh fedora-eln-boot done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-boot-20231129-184506

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-boot-20231129-184545.log

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231129-190101.log

**Comment by cockpituous** [comment]
image-refresh fedora-39 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-39-20231129-192718

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-39-20231129-192727.log

**Comment by martinpitt** [comment]
That ELN failure was transient, retrying.

File: images/scripts/fedora.setup [hunk 7]
```diff
@@ -220,8 +240,10 @@ rm -rf /var/log/audit/
 # clean up old kernels after upgrading
 dnf repoquery --installonly --latest-limit=-1 -q | xargs --no-run-if-empty rpm -e --verbose
 
-# pmlogger.service sometimes causes long shutdown hangs; disable all PCP services and timers
-systemctl --all --legend=false list-units 'pm*' | awk '{print $1}' | xargs systemctl disable
+if [ "fedora-eln" != "$IMAGE" ]; then
+    # pmlogger.service sometimes causes long shutdown hangs; disable all PCP services and timers
+    systemctl --all --legend=false list-units 'pm*' | awk '{print $1}' | xargs systemctl disable
+fi
```
Comment (martinpitt) [inline images/scripts/fedora.setup:246]: Actually.. [that recent build failure](https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231129-190101.log) is exactly the reason why we need this hack. Please put it back for ELN. In fact, it seems it needs to become stronger, and append a `--now`.

**Review by martinpitt — COMMENTED** [review]
Let's see if the retry works -- if so, then the fix below can become a follow-up.

Do these images work for you now? Do you need some help with this?

**Comment by cockpituous** [comment]
image-refresh fedora-eln done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-20231130-064802

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231130-064848.log

File: images/scripts/fedora.setup (hunk 7 above)
Comment (KKoukiou) [reply images/scripts/fedora.setup:246]: I had it in place and if I recall there were not units matched, making xargs run with no arguments and fail.
Retrying it again.

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-boot-20231130-080606.log

File: images/scripts/fedora.setup (hunk 7 above)
Comment (martinpitt) [reply images/scripts/fedora.setup]: Nice, thank you!

**Comment by cockpituous** [comment]
image-refresh fedora-eln done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-20231130-082542

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231130-082525.log

**Comment by cockpituous** [comment]
image-refresh fedora-39 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-39-20231130-084602

**Comment by cockpituous** [comment]
image-refresh fedora-eln-boot done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-boot-20231201-110923

**Comment by cockpituous** [comment]
image-refresh fedora-eln done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-20231201-112653

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231201-112626.log

**Comment by martinpitt** [comment]
[that failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-5585-20231201-193416-e17da9f0-fedora-eln-boot-anaconda-webui-pr-70-rhinstaller-anaconda-webui/log.html) should be fixed by https://github.com/rhinstaller/anaconda-webui/pull/75 , retrying

**Comment by martinpitt** [comment]
I rebased this, as the previous version had [outdated images](https://cockpit-logs.us-east-1.linodeobjects.com/pull-5585-20231205-114736-e17da9f0-fedora-eln-boot-anaconda-webui-pr-70-rhinstaller-anaconda-webui/log.html)

**Review by martinpitt — APPROVED** [review]
This LGTM. As this doesn't break anything, I'm happy to land this even though https://github.com/rhinstaller/anaconda-webui/pull/70 isn't ready yet. This keeps getting stale, so please either land it now, or (if it's too broken and not relevant), let's shelve it.

**Comment by KKoukiou** [comment]
It's relevant - I just was hoping to get at least a semi-successful test run before merging.

**Comment by KKoukiou** [comment]
Now it fails again with some other error, I will check

**Comment by cockpituous** [comment]
image-refresh fedora-eln-boot done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-boot-20231205-130136

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-boot-20231205-130101.log

**Comment by cockpituous** [comment]
image-refresh fedora-eln done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-eln-20231205-131831

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-eln-20231205-131818.log

**Review by martinpitt — APPROVED** [review]
Thanks!

I am trying to reproduce the -boot image failures locally. Unfortunately that image doesn't set up a serial console, so `virsh console` doesn't work, and thus we cannot capture/log boot failures.

I'll see what's going on with virt-viewer, once I get a boot failure locally.

**Comment by martinpitt** [comment]
I land this now. The fedora-eln image works reasonably well, and the -boot image is just a simple curl anyway, no modifications. So if it's broken, it needs to be fixed in Fedora. It could also be that it requires different QEMU options on our side, I'll debug that this morning in https://github.com/rhinstaller/anaconda-webui/pull/70

**Pull request merged**